pkg/controller/network/networkinterface/managed.go

```diff
@@ -0,0 +1,147 @@
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-azure/apis/network/v1alpha3"
+	azureclients "github.com/crossplane/provider-azure/pkg/clients"
+	"github.com/crossplane/provider-azure/pkg/clients/network"
+)
+
+// Error strings.
+const (
+	errNotNetworkInterface    = "managed resource is not an NetworkInterface"
+	errCreateNetworkInterface = "cannot create NetworkInterface"
+	errGetNetworkInterface    = "cannot get NetworkInterface"
+	errDeleteNetworkInterface = "cannot delete NetworkInterface"
+)
+
+// Setup adds a controller that reconciles Public Ip Address.
+func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(v1alpha3.NetworkInterfaceGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha3.NetworkInterface{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha3.NetworkInterfaceGroupVersionKind),
+			managed.WithConnectionPublishers(),
+			managed.WithExternalConnecter(&connecter{client: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connecter struct {
+	client client.Client
+}
+
+func (c *connecter) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	creds, auth, err := azureclients.GetAuthInfo(ctx, c.client, mg)
+	if err != nil {
+		return nil, err
+	}
+	cl := azurenetwork.NewInterfacesClient(creds[azureclients.CredentialsKeySubscriptionID])
+	cl.Authorizer = auth
+	return &external{client: cl}, nil
+}
+
+type external struct {
+	client networkapi.InterfacesClientAPI
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	s, ok := mg.(*v1alpha3.NetworkInterface)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotNetworkInterface)
+	}
+
+	az, err := e.client.Get(ctx, s.Spec.ResourceGroupName, meta.GetExternalName(s), "")
+	if azureclients.IsNotFound(err) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetNetworkInterface)
+	}
+
+	network.UpdateNetworkInterfaceStatusFromAzure(s, az)
+	s.SetConditions(xpv1.Available())
+
+	o := managed.ExternalObservation{
+		ResourceExists:    true,
+		ConnectionDetails: managed.ConnectionDetails{},
+	}
+
+	return o, nil
```

Comment:
```suggestion
	return managed.ExternalObservation{
		ResourceExists:    true,
		ResourceUpToDate:  true,
	}, nil
```
We return `ResourceUpToDate: true` because I see that nothing can be updated in networkinterface.external.Update. Is this really the case, can no fields be updated? What about tags?